src/op_mode/qos.py

```diff
@@ -0,0 +1,227 @@
+                    tmp['requeues'] = classes.get('stats', {}).get('requeues', 0)
+                    tmp['lended'] = classes.get('stats', {}).get('lended', 0)
+                    tmp['borrowed'] = classes.get('stats', {}).get('borrowed', 0)
+                    tmp['giants'] = classes.get('stats', {}).get('giants', 0)
+
+                    output_dict['classes'][class_name] = tmp
+                    raw_dict['qos'][interface_name][class_name] = tmp
+
+        # Skip printing of values for this interface. All interfaces will be returned in a single dictionary if 'raw' is called
+        if raw:
+            continue
+
+        # Default class may be out of order in original JSON. This moves it to the end
+        move_default = output_dict.get('classes', {}).pop('default', None)
+        if move_default:
+            output_dict.get('classes')['default'] = move_default
+
+        # Create the tables for outputs
+        for output in output_dict.get('classes'):
+            data = output_dict.get('classes').get(output)
+
+            # Add values for detailed (list view) output
+            if detail:
+                output_list.append([data['interface_name'],
+                                    data['policy_name'],
+                                    data['direction'],
+                                    data['class_name'],
+                                    data['queue_type'],
+                                    data['rate'],
+                                    data['ceil'],
+                                    data['bytes'],
+                                    data['packets'],
+                                    data['drops'],
+                                    data['queued'],
+                                    data['overlimits'],
+                                    data['requeues'],
+                                    data['lended'],
+                                    data['borrowed'],
+                                    data['giants']]
+                                    )
+            # Add values for normal (table view) output
+            else:
+                output_list.append([data['class_name'],
+                                    data['queue_type'],
+                                    format_data_type(int(data['rate']), 'b'),
+                                    format_data_type(int(data['ceil']), 'b'),
+                                    format_data_type(int(data['bytes']), 'B'),
+                                    data['packets'],
+                                    data['drops'],
+                                    data['queued']]
+                                    )
+
+        if output_list:
+            if detail:
+                # Headers for detailed (list view) output
+                headers = ['Interface', 'Policy Name', 'Direction', 'Class', 'Type', 'BW', 'Max. BW', 'Bytes', 'Pkts.', 'Drops', 'Queued', 'Overlimit', 'Requeue', 'Lended', 'Borrowed', 'Giants']
```

Comment:
- Delete the dot in the `Pkts`
- Rename `BW` to `Bandwidth`

Can you add at least one new line between interfaces?
```
vyos@r4# run show qos shaping 
--------------------------------------------------------------------------------
Interface: eth0
Policy Name: SHAPE

Class    Type          BW    Max. BW      Bytes    Pkts.    Drops    Queued
-------  ------  --------  ---------  ---------  -------  -------  --------
root     htb     1.000 Gb   1.000 Gb  14.098 KB       99        0         0

--------------------------------------------------------------------------------
Interface: eth1
Policy Name: SHAPE

Class    Type          BW    Max. BW    Bytes    Pkts.    Drops    Queued
-------  ------  --------  ---------  -------  -------  -------  --------
root     htb     1.000 Gb   1.000 Gb     0  B        0        0         0

[edit]
vyos@r4# 
```

Reply:
> Does the below look like what you're asking for regarding the spacing? Increase the spacing between interfaces to 2 lines?

Yes, it does. It looks better.

Reply:
The extra return line between interfaces didn't take the first time. It seems op mode eats any empty line, so I needed to add padding to the line to make it work.